huey/backends/redis_backend.py

```diff
@@ -78,12 +78,25 @@ def read(self):
             return None
 
 
+# a custom lua script to pass to redis that will read tasks from the schedule
+# and atomically pop them from the sorted set and return them.
+# it won't return anything if it isn't able to remove the items it reads.
+SCHEDULE_POP_LUA = """
+local key = KEYS[1]
+local unix_ts = ARGV[1]
+local res = redis.call('zrangebyscore', key, '-inf', unix_ts)
+if #res and redis.call('zremrangebyscore', key, '-inf', unix_ts) == #res then
+    return res
+end
```

Comment:
that gives you the count of the elements it removed from the sorted set. If the count doesn't match up with the number you intended, that means it wasn't able to remove those elements from the sorted set and you don't want to return those elements from the list to be worked on. It will almost never happen since redis will be working inside that single thread to do that operation atomically but still it's not a bad practice to checksum it and it won't introduce any latency.